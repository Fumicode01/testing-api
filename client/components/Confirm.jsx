import React, { useContext, useEffect } from 'react';
import { AppContext } from '../context/AppContext';
import { useRouter } from 'next/router'
import axios from 'axios';
import Typography from '@mui/material/Typography';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

const config = {
    headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer qhtfs87hjnc12kkos',
    }
  
}



export const Confirm = ({ activeStep, setActiveStep, handleBack,  steps })  => {
    const { state, dispatch } = useContext(AppContext);
    const { items, shipping, consumer, totalAmount } = state
    const router = useRouter()
    
    useEffect(()=>{
        console.log(state)
    },[])

    const handleClick = async () => {
       const res = await axios.post('http://localhost:5000/api/order', state, config)
        try{
            console.log(res.data)
            // setActiveStep(activeStep + 1);
            router.push(res.data.checkoutUrl)
        } catch(err){
            console.error(err)
        }
     
    }

  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom>
        Order summary
      </Typography>
      <List disablePadding>
        {items.map((item) => (
          <ListItem key={item.name} sx={{ py: 1, px: 0 }}>
            <ListItemText primary={item.name} />
            <Typography variant="body2">{item.price.amount}</Typography>
          </ListItem>
        ))}

        <ListItem sx={{ py: 1, px: 0 }}>
          <ListItemText primary="Total" />
          <Typography variant="subtitle1" sx={{ fontWeight: 700 }}>
              {totalAmount.amount}
          </Typography>
        </ListItem>
      </List>
      <Grid container spacing={2}>
        <Grid item xs={12} sm={6}>
          <Typography variant="h6" gutterBottom sx={{ mt: 2 }}>
            User Info
          </Typography>
          <Typography gutterBottom>{consumer.givenNames} {consumer.sername}</Typography>
          <Typography gutterBottom>{consumer.email}</Typography>
          {consumer.phoneNumber && <Typography gutterBottom>{consumer.phoneNumber}</Typography>}

        </Grid>
        <Grid item container direction="column" xs={12} sm={6}>
          <Typography variant="h6" gutterBottom sx={{ mt: 2 }}>
            Shipping Detail
          </Typography>
          <Typography gutterBottom>{shipping.line1}</Typography>
          <Typography gutterBottom>{shipping.suburb}</Typography>
          <Typography gutterBottom>{shipping.postcode}</Typography>
          <Typography gutterBottom>{shipping.name}</Typography>
        </Grid>
        <Box sx={{ display: 'flex', justifyContent: 'flex-end' }}>
                {activeStep !== 0 && (
                    <Button onClick={handleBack} sx={{ mt: 3, ml: 1 }}>
                    Back
                    </Button>
                )}

                <Button
                    variant="contained"
                    onClick={handleClick}
                    sx={{ mt: 3, ml: 1 }}
                >
                    {activeStep === steps.length - 1 ? 'Place order' : 'Next'}
                </Button>
            </Box>
      </Grid>
    </React.Fragment>
  );
}