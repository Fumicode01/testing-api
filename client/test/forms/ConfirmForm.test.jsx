import React from 'react';
import axios from 'axios'
import { render, screen, cleanup, fireEvent, waitForElement, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Confirm } from '../../components/Confirm';
import { axiosErrorResponse400, axiosErrorResponse401 } from '../mocks/mockData'


jest.mock('axios')
afterEach(cleanup);
afterAll(jest.clearAllMocks);

test('Component should be rendered', async() => {
    render(<Confirm steps={2}/>);
    const orderButton = screen.getByTestId("confirm-order");
    expect(orderButton).toBeInTheDocument("PLACE ORDER");
})

test('Component should be rendered', async() => {
    render(
        <Confirm steps={2} />
    );
    const text = screen.getByText("Order summary");
    expect(text).toBeInTheDocument();
})

// test('The Confirm Order button has been clicked', async() => {
//     const handleClick = jest.fn();
//     render(<Confirm steps={2} handleClick={handleClick} />);
//     const orderButton = screen.getByTestId("confirm-order");
//     fireEvent.click(orderButton);

//     expect(handleClick).toHaveBeenCalled();
// })

describe("Sned API request", () => {
    // describe("When API call is successful", () => {
    //     const AxiosResponse = {
    //         data:{
    //             "token": "01L0FW6CT5",
    //             "expires": "2022-03-06T23:39:42.905Z",
    //             "checkoutUrl": "https://portal.staging.scalapay.com/checkout/01L0FW6CT5"
    //         }
    //     }
    //     jest.spyOn(axios, "post").mockResolvedValue(AxiosResponse);
    //     render(<Confirm steps={2} />);
    //     expect(axiosMock.post).toHaveBeenCalledTimes(1);

    // })
        // test("When API call not success", async () => {
        //     jest.spyOn(axios, 'post').mockRejectedValue(()=>({}));
        //     render(<Confirm steps={2} />)
        //     const error = screen.getByTestId("error");
        //     await waitFor(() => expect(error).toBeInTheDocument())
        // })
        test("Input values are not suit", async () => {
            await jest.spyOn(axios, "post").mockResolvedValue(axiosErrorResponse400);
            render(<Confirm steps={2} />);
            const error = screen.getByText('Please check your details and try again.')
            await waitFor(() => expect(error).toBeInTheDocument())
        })

        test("Authorization error", async () => {
            await jest.spyOn(axios, "post").mockResolvedValue(axiosErrorResponse401);
            render(<Confirm steps={2} />);
            const error = screen.getByTestId("error");
            // const error = screen.getByText('Authentication failed. Please login again or Contact us.')
            await waitFor(() => expect(error).toBeInTheDocument())
        })


})