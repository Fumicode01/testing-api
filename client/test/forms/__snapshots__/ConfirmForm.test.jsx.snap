// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirm /> When Confirm component has been rendered with global state from Context Create snapshot of Confirm component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        Order summary
      </h6>
      <ul
        class="MuiList-root css-1mk9mw3-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              iPhone X
            </span>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            100
          </p>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Total
            </span>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-1isisa7-MuiTypography-root"
          >
            100
          </h6>
        </li>
      </ul>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
          >
            User Info
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
             
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1h2i2a7-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
          >
            Shipping Detail
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-1bvc4cc"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1dym4ij-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Back
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-8bjq0v-MuiButtonBase-root-MuiButton-root"
            data-testid="confirm-order"
            tabindex="0"
            type="button"
          >
            Place order
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        Order summary
      </h6>
      <ul
        class="MuiList-root css-1mk9mw3-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              iPhone X
            </span>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            100
          </p>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Total
            </span>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-1isisa7-MuiTypography-root"
          >
            100
          </h6>
        </li>
      </ul>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
          >
            User Info
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
             
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1h2i2a7-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
          >
            Shipping Detail
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-1bvc4cc"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1dym4ij-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Back
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-8bjq0v-MuiButtonBase-root-MuiButton-root"
            data-testid="confirm-order"
            tabindex="0"
            type="button"
          >
            Place order
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
    >
      Order summary
    </h6>
    <ul
      class="MuiList-root css-1mk9mw3-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            iPhone X
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          100
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-den40q-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Total
          </span>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1isisa7-MuiTypography-root"
        >
          100
        </h6>
      </li>
    </ul>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
        >
          User Info
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
           
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1h2i2a7-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-xzs2lf-MuiTypography-root"
        >
          Shipping Detail
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        />
      </div>
      <div
        class="MuiBox-root css-1bvc4cc"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1dym4ij-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Back
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-8bjq0v-MuiButtonBase-root-MuiButton-root"
          data-testid="confirm-order"
          tabindex="0"
          type="button"
        >
          Place order
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
